src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -0,0 +1,501 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+#include "common.h"
+
+#include <windows.h>
+
+#include "CommonTypes.h"
+#include "CommonMacros.h"
+#include "daccess.h"
+#include "PalRedhawkCommon.h"
+#include "regdisplay.h"
+#include "ICodemanager.h"
+#include "CoffNativeCodeManager.h"
+#include "varint.h"
+
+#define UBF_FUNC_KIND_MASK      0x03
+#define UBF_FUNC_KIND_ROOT      0x00
+#define UBF_FUNC_KIND_HANDLER   0x01
+#define UBF_FUNC_KIND_FILTER    0x02
+
+#define UBF_FUNC_HAS_EHINFO     0x04
+
+#if defined(_TARGET_AMD64_)
+
+//
+// Define unwind code structure.
+//
+typedef union _UNWIND_CODE {
+    struct {
+        uint8_t CodeOffset;
+        uint8_t UnwindOp : 4;
+        uint8_t OpInfo : 4;
+    };
+
+    uint16_t FrameOffset;
+} UNWIND_CODE, *PUNWIND_CODE;
+
+//
+// Define unwind information flags.
+//
+
+#define UNW_FLAG_NHANDLER 0x0
```

Comment:
Can you please add comment on what each of these flags mean? I don't know what is EHANDLER / UHANDLER / NHANDLER.